src/common/utils/playerDynamics.ts

```diff
@@ -57,32 +66,47 @@ type PlayerCollisionParams = {
   boundsTree?: MeshBVH;
 };
 
+const GlobalCollisionsObjects = {
+  collisionRay: new Line3(),
+  playerCollisionBoundingBox: new Box3(),
+  closestTrianglePoint: new Vector3(),
+  closestPlayerRayPoint: new Vector3(),
+  collisionDisplacementUnitVector: new Vector3(),
+  collisionDisplacementVector: new Vector3(),
+};
+
 /**
  * Determine how to adjust the player when colliding with an object
  * @param params PlayerCollisionParams detailing required components to calculate player collision
  * @returns A unit vector pointing in the direction of adjustment if a collision is registered
  */
 export const calculatePlayerCollisions = (
   params: PlayerCollisionParams,
-): Vector3 => {
+  target: Vector3,
+): void => {
```

Comment:
Shouldn't be necessary to explicitly type as `void`

```suggestion
) => {
```